Add util logger for supervisor to log to orchestrator

lrlakshan opened the pull request:
@isoteemu please review

heikkilv commented:
I did not test your changes, but looking at the changes done. It might be beneficial to have some logging done also on the other routes defined in app.py. For example, when Wasm modules are uploaded/updated to the supervisor, as well as when some function is called in the existing Wasm modules.

isoteemu reviewed (changes requested):
- I'm **guessing** that the `logging.handlers.HTTPHandle` is blocking. If it is, please create an issue to convert it to non-blocking, as it can seriously affect the performance. For now blocking is fine.
- It should integrate to the default logging handler. Integrate it in the `host_app/flask_app/logging/logger.py`. In code any `host_app` message sent with `logger.info|warning|debug` etc should be sent to a log consumer.
    - Watch out that when logging is sent to debug and sending logs that the http client messages do not trigger another sending event ;).
- `remote_addr` returns the client address. It can be other than orhestrator, such as another wasmiot supervisor, or direct consumer/actor. I am not sure what the ideal solution for implementation is. But I'm feeling that if the logging endpoint is orchestrator, it should be detected on a deployment basis. Meaning when running wasm-module deployment, logs should be sent __by default__ to orchestrator that made the deployment, and general logs to the orchestrator authenticated with supervisor. Currently there is no authentication mechanism, so it's just the orchestrator found with mdns. All that said, it should also be configurable. Having it as a configuration variable is sufficient for now, and the endpoint should be set in some config variable like `WASMIOT_LOGGING_ENDPOINT`. <https://flask.palletsprojects.com/en/2.3.x/config/>. Note that we rely on [environment for configuration](https://12factor.net/config), so it needs extra steps to check if the config is set, and using likes of `app.config.setdefault()` to populate it from the environment.
- Log messages are better to be sent as likes of structured json strings. Formatter converts messages into json strings, and in the orchstrator then actually formats the strings. This way it can include additional structured context information in future, such as traces, `extra` and such. There __might__ be a common format already for those.
    - The `message` itself should be formatted in python.
- Please include docstrings in functions, and preferably type hints. [Function documentation works a bit differently in python](https://realpython.com/documenting-python-code/#documenting-your-python-code-base-using-docstrings). We have not decided on a specific format, but we seem to have used mostly reStructuredText (rst) format.

lrlakshan commented:
Thanks for the feedback. I will work on the mentioned suggestions.

lrlakshan commented:
- As the logging.handlers.HTTPHandle is blocking, added a logging queue handler and hope that will help.
- Added configurable WASMIOT_LOGGING_ENDPOINT add the orchestrator endpoint
- Log messages sending as structured json strings. 
- Add docstrings for logger.py
